pkg/system/db_reconciler.go

```diff
@@ -135,52 +233,115 @@ func (r *Reconciler) reconcileCluster() error {
 	}
 
 	// update db volume resources
-	desiredStorageConfiguration, err := getDesiredStorageConfiguration(dbSpec)
+	// update the storage configuration
+	err := setDesiredStorageConf(&r.CNPGCluster.Spec.StorageConfiguration, dbSpec)
 	if err != nil {
-		r.Logger.Errorf("got error getting desired storage configuration for cnpg cluster. error: %v", err)
+		r.cnpgLogError("got error getting desired storage configuration for cnpg cluster. error: %v", err)
 		return err
 	}
-	r.CNPGCluster.Spec.StorageConfiguration = desiredStorageConfiguration
 
 	// TODO: consider specifying a separate WAL storage configuration in Spec.WalStorage
 	// currently, the same storage will be used for both DB and WAL
 
-	// TODO: handle import of existing DB in bootstrap section
-	// define bootstrap section
-	r.CNPGCluster.Spec.Bootstrap = &cnpgv1.BootstrapConfiguration{
-		InitDB: &cnpgv1.BootstrapInitDB{
-			Database: "nbcore",
-			Owner:    "noobaa",
-		},
-	}
+	return nil
 
-	// check if the cluster is changed
-	if reflect.DeepEqual(*existingClusterSpec, r.CNPGCluster.Spec) {
-		// The cluster spec is unchanged, no need to update
-		return nil
+}
+
+func (r *Reconciler) reconcileClusterImport() error {
+
+	// The bootstrap configuration should only be set for the CR creation.
+
```

Comment:
regardless of this code only running in cluster creation flow. guard it to look at the noobaaDbCluster